BeagleBone_black/src/main_threading.cpp

```diff
@@ -20,37 +20,73 @@
  *    limitations under the License.
  */
 
-#include <iostream>
+// #include <iostream>
+#include <stdio.h>
 
-#include "utils/concurrent/Thread.hpp"
+#include "utils/concurrent/thread.hpp"
+#include "utils/concurrent/lock.hpp"
+#include "utils/concurrent/condition_variable.hpp"
 
 using hyped::utils::concurrent::Thread;
+using hyped::utils::concurrent::Lock;
+using hyped::utils::concurrent::ScopedLock;
+using hyped::utils::concurrent::ConditionVariable;
 
-void foo()
+ConditionVariable cv;
+Lock              global_lock;
+Lock              lock_;
+
+int value = 0;
+
+void delay(uint32_t i)
 {
-  std::cout << "New foo thread started" << std::endl;
+  while (i--);
 }
 
 class DemoThread: public Thread {
  public:
-  explicit DemoThread(uint8_t id): Thread(id) { /* EMPTY */ }
+  explicit DemoThread(uint8_t id, bool synchronise = false)
+    : Thread(id),
```

Comment:
There should be a 4 space indent here. See https://hyp-ed.github.io/styleguide/#Constructor_Initializer_Lists